src/task/future.rs

```diff
@@ -151,6 +164,47 @@ unsafe fn clone_task(task: *const Task) -> TaskCell {
     task_cell
 }
 
+thread_local! {
+    /// Local queue reference that is set before polling and unset after polled.
+    static LOCAL: Cell<*mut Local<TaskCell>> = Cell::new(std::ptr::null_mut());
+}
+
+unsafe fn wake_task(task: Cow<'_, Arc<Task>>, reschedule: bool) {
+    LOCAL.with(|ptr| {
+        if ptr.get().is_null() {
+            // It's out of polling process, has to be spawn to global queue.
+            // It needs to clone to make it safe as it's unclear whether `self`
+            // is still used inside method `spawn` after `TaskCell` is dropped.
```

Comment:
Strictly speaking, there is still some rism because `Waker` is accidentally `Sync`. https://github.com/rust-lang/rust/issues/66481
It means the user might be able to move the reference to other threads (for example, using crossbeam::scoped`) and break our assumption.

However, I think it can be ignored. Maybe we can give some documentation and use `expect` to give some messages?